src/v2/blobs.rs

```diff
@@ -100,6 +102,12 @@ impl Client {
                     )))
                 }
             });
+
+        let fres = fres_digest.join(fres_reqwest).and_then(|(digest, body)| {
```

Comment:
`s/join/and_then/` maybe? As in, don't get to `reqwest` if the input-digest is known to be malformed?

Reply:
While looking at the documented of `join` is not fully conclusive, [it is documented to drop the other future if one fails](https://docs.rs/futures/0.1.26/futures/future/trait.Future.html#method.join). Exploring your suggestion, the `join` can't be simply substituted by `and_then` which expects an `FnOnce -> IntoFuture` (simplified) as its argument as opposed to an `IntoFuture`, so it would complicate things more by nesting a level.

Looking at the macro which generates the `join` method, reveals that it [polls the operands in the given order and fails early](https://github.com/rust-lang-nursery/futures-rs/blob/043b16f96732c07df560e6198728ec470d7e96e6/src/future/join.rs#L66-L82), so I think we're good here.

Reply:
Ack, let's stick to `join` then.